src/Workspaces/Core/Portable/Shared/Utilities/StringBreaker.cs

```diff
@@ -275,13 +275,14 @@ private static TextSpan ScanWordRun(string identifier, int length, int wordStart
                 {
                     // hit the 'o' in XMLDo.  Return "XML"
                     Debug.Assert(char.IsUpper(identifier[current - 1]));
-                    return TextSpan.FromBounds(wordStart, current - 1);
```

Comment:
agreed.  But it also has increased perf overhead (which showed up in my traces).  It's arguable that FoundBounds should not do any checks and should just defer to the constructor to do them.